cmd/kubectl-gadget/deploy.go

```diff
@@ -336,17 +350,23 @@ func runDeploy(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("it's not possible to use --quiet and --debug together")
 	}
 
-	objects, err := parseK8sYaml(resources.GadgetDeployment)
-	if err != nil {
-		return err
-	}
+	if useSeccompProfile {
+		resourcesYAML = append(resourcesYAML[:1], resourcesYAML...)
 
-	traceObjects, err := parseK8sYaml(resources.TracesCustomResource)
-	if err != nil {
-		return err
+		// We need to add the secomp profile creation after the namespace because it
+		// lives in the gadget namespace.
+		resourcesYAML[1] = resources.GadgetSeccompProfile
```

Comment:
The append() and replacement is confusing.

If `resourcesYAML` was a `[]*string`, you could just set the item to nil:
```
resourcesYAML[resourcesSeccompIndex] = nil
```

And add a `continue` in the loop below if `resource==nil`.